### ISSUE TITLE
Append mode together with mapping table may have an issue

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-pytestlog2db#12]
The pytest for PythonExtensionsCollection contains 131 tests. The PyTestLog2DB configuration file contains the following entries:

```
   "version_sw" : "apptest-004",
   "variant"    : "QT-004"
```
But no matching mapping rule!

Outcome: In test database all 131 tests are counted properly; visible under: QT-004 / apptest-004 / All Components

I compute the same pytest result file again, but now in append mode.

Outcome: In test database now 262 tests are counted properly; still visible under: QT-004 / apptest-004 / All Components

OK so far.

I changed the configuration to

   "version_sw" : "apptest-005",
   "variant"    : "QT-005"

Additionally I added a matching mapping rule:

"PythonExtensionsCollection" : "Comparison"

This mapping will hit 10 tests (out of the 131 available tests). With this setting I repeated the PyTestLog2DB execution (with still the same pytest XML result file).

Outcome (after first run):

In test database 10 tests are counted properly; visible under: QT-005 / apptest-005 / PythonExtensionsCollection

Outcome (after second run in append mode):

In test database still 10 tests are counted; still visible under: QT-005 / apptest-005 / PythonExtensionsCollection

This I do not understand. I would have expected 20 tests counted.

Switching to view

QT-005 / apptest-005 / All Components

I again get 262 tests counted (like expected)

I can also see that the "component" drop down list contains the entry "PythonExtensionsCollection" two times (and also "unknown" two times). How can this be that such a list contains duplicate entries?

**Comment by HolQue** [comment]
Now I understand:

Every "append" adds another entry of the same component name in the component selection box of the dashoard. But the number of tests are not increased. This means: The values are not accumulated under one common entry in the component selection box of the dashoard, but still separated and selectable by individual entries in the component selection box.

This happens only in case of a mapping table is involved.

**Comment by test-fullautomation** [comment]
Hi Ngoan,

please verify. Looks like a bug.

Thank you,
 Thomas

**Comment by ngoan1608** [comment]
Hi Holger,
Hi Thomas,

I have tried to reproduce this issue on my local webapp and database but everything is fine.
I checked db and webapp, the data is updated properly.

I take a investigation today I found a potential root cause of Holger's issue is _**the stored procedure `update_evtbl` is missing in your database**_.
Could you help me to double check it on your MySQL WorkBench?

Thank you,
Ngoan

**Comment by HolQue** [comment]
OK, what do I have to do?

**Comment by test-fullautomation** [comment]
Hi Ngoan,
maybe we can check together tomorrow.
Thank you,
 Thomas

**Comment by ngoan1608** [comment]
Hi Thomas,
Hi Holger,

As my explanation in the email, this is the issue with **testdb** that Holger used for importing.
The **testdb** is not updated with changes in _evtbl__ table and store procedure to avoid the duplicated component.

I have updated **testdb** with latest changed and rebuilt data in the _evtbl__ table. The result is displayed properly on webapp now.

This ticket can be closed.

Thank you,
Ngoan

**Comment by HolQue** [comment]
Values displayed properly now.

(retest with latest available feature branches / 28.02.2023)
